include/blmc_robots/real_finger.hpp

```diff
@@ -373,4 +388,188 @@ robot_interfaces::finger::BackendPtr create_real_finger_backend(
     return backend;
 }
 
+class RandomFingerDriver : public robot_interfaces::RobotDriver<
+                              robot_interfaces::finger::Action,
+                              robot_interfaces::finger::Observation>
+{
+
+public:
+    typedef robot_interfaces::finger::Action Action;
+    typedef robot_interfaces::finger::Observation Observation;
+    typedef robot_interfaces::finger::Vector Vector;
+    typedef robot_interfaces::finger::Status Status;
+
+    //default constructor, error: implicitly deleitng
+    RandomFingerDriver() : RobotDriver(0.003, 0.005) {}
+
+    int i = 0;
+
+    Observation get_latest_observation() override {
+
+      Observation observation;
+      observation.angle[0] = i;  observation.angle[1] = 2*i; observation.angle[2] = 3*i;
+      observation.velocity[0] = i + 1; observation.velocity[1] = 2*i + 1; observation.velocity[2] = 3*i + 1;
+      observation.torque[0] = i + 2; observation.torque[1] = 2*i + 2; observation.torque[2] = 3*i + 2;
+
+      i++;
+
+      return observation;
+
+    }
+
+    //check for better "void" implementations
+
+    Action apply_action(const Action &desired_action) override {
+
+      // (maybe) isolated casue to this: the random driver just goes on spitting values,
+      //  and by the time one observation is written to the file, 10 newer ones get
+      // added to the buffer already. View this by printing out the index and the
+      // logger_data_->observation->newest_timeindex_()
+      double start_time_sec = real_time_tools::Timer::get_current_time_sec();
+
+      real_time_tools::Timer::sleep_until_sec(start_time_sec + 0.0001);
```

Comment:
I forgot about this issue.  Can you try with `+ 0.001` again (to have the same rate as the actual robot)?  If it is still failing then, this implies that there is an issue somewhere that we should find before merging.